components/camel-tahu/src/test/java/org/apache/camel/component/tahu/TahuHostConsumerTest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.tahu;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.test.infra.core.annotations.RouteFixture;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Disabled
+public class TahuHostConsumerTest extends TahuTestSupport {
+
+    static enum HostAppTestProfile {
+        SESSION_ESTABLISHMENT_TEST("host SessionEstablishmentTest", false),
+        SESSION_TERMINATION_TEST("host SessionTerminationTest TestHostApp", true),
+        // SEND_COMMAND_TEST("host SendCommandTest G2 E2 D2", false),
+        // EDGE_SESSION_TERMINATION_TEST("host EdgeSessionTerminationTest G2 E2 D2", false),
+        // MESSAGE_ORDERING_TEST("host MessageOrderingTest G2 E2 D2 5000", false),
+        ;
+
+        private HostAppTestProfile(String testConfig, boolean disconnect) {
+            this.testConfig = testConfig;
+            this.disconnect = disconnect;
+        }
+
+        final String testConfig;
+        final boolean disconnect;
+    }
+
+    private static final Logger LOG = LoggerFactory.getLogger(TahuHostConsumerTest.class);
+
+    @ParameterizedTest
+    @EnumSource
+    public void tckSessionTest(HostAppTestProfile profile) throws Exception {
+
+        initiateTckTest(profile.testConfig);
+
+    }
+
+    @RouteFixture
```

Comment:
(advisory) IMHO, these sets are simple enough that they can use the traditional `CamelTestSupport` from `camel-test-junit-5`. 
Typically, we use the `camel-test-infra-core` for components that have *very* complex testing needs/patterns (i.e.; `camel-jms`, `camel-kafka`) ... because its flexibility comes at the cost of an API that is much harder to use.

Reply:
I really tried to pare down the test code. Every time I tried to pull away from `test-infra` the tests would stop working altogether.

Totally understand the API is more challenging but I didn't find it much harder than using Jupiter extensions.

If you don't mind, could we please leave these as they are?

Reply:
Yes, it's OK to use it as is. It's pretty neat, after the cleanup, actually.